### ISSUE TITLE
joint probability estimation

### ISSUE THREAD

**Issue by IQ17** [dluvizon/pose-regression#5]
Hi, thanks for the paper!
Many paper discussed about how to "infer" occluded joints using contextual info, but none of them estimate visibility probability. 

I am working on estimating the visibility of a joint, to avoid output unreliable (x, y) on occluded joints. But I am not so sure if sigmoid of the max value of heat map is a good way, as in order to give low probability, the max value should be negative, which is a very strong regularization on heat maps. By training with this sigmoid cross entropy loss, I found the predicted (x, y) became worse. 

Could you justify your method a bit more, and provide some experimental evidence or discussion on its preference? 

Thansk!

**Comment by dluvizon** [comment]
Hello,
The sigmoid represents the "strength" of heatmaps, but do not guarantee the "pointness" of heatmaps.
It is possible that the "pointness" of such heatmaps will be more informative.
On the other hand, I do not see any problem by enforcing negative values in the heatmaps, since it will be normalized before the sotf-argmax (in our approach).

We did not observed a significant change in performance when not considering supervision on visibility.

**Comment by IQ17** [comment]
Thanks for the explanation! 

I totally agree with you that enforcing negative values in the heatmaps is ok. 
However, I did several experiments, comparing with/without sigmoid cross-entropy loss, on my dataset, and consistently got 2% PCKh@0.2 better when training without the loss. When training without this loss, I also drawed its curve and found its value went high during optimisation and became stable at a high level. This phenomenon indicates that enforcing negative values in the heatmaps is a constraint.

More importantly, the false positive rate is too high... Is this probability only used for contextual info aggregation, not for directly output?

**Comment by dluvizon** [comment]
> More importantly, the false positive rate is too high... Is this probability only used for contextual info aggregation, not for directly output?

It may depends on what you consider as a false positive.
Our method was trained considering visible joints even the ones that are occluded. We only set "non-visible" to the joints without ground truth coordinates (x,y), or with coordinates outside the crop.
We supervise the probability directly, but using a coefficient of 0.01, since the cross entropy is much stronger then the L1 norm on coordinates.

**Comment by IQ17** [comment]
I think I wrongly used the sigmoid idea. Below are my opinions.
This idea is very good for estimating visibility/invisibility keypoints. 
But, it should not be used for estimating existing/non-existing keypoints.

Visibility keypoints and existing keypoints can be thought as the same.
The difference between "invisibility" and "non-existing" is the following:
1) invisibility is for describing keypoints physically inside the image, but are occluded or self-occluded. As keypoints are inside the image, its contextual info are rich;
2) non-existing is for describing keypoints physically outside the image, e.g. when only the upper-half body is photoed, knees and ankles are non-existing keypoints, and they have little contextual info;

So when the dataset contains many half-body images, there will be lots of non-existing keypoints, and the performance of model with trained sigmoid loss will be worse than model trained without sigmoid loss.
